src/development/troubleshoot.md

```diff
@@ -161,3 +135,75 @@ where `<project_id>` is the random-character ID of the project.  That can be fou
 If you see a bare "File not found" error when accessing your Drupal site with a browser, this means that you've pushed your code as a vanilla project but no *index.php* has been found.
 
 Make sure your repository contains an *index.php* file in the [web location root](/configuration/app-containers.md#locations), or that your [Drush](/frameworks/drupal7/drush.md) make files are properly named.
+
+
+## Stuck build (or deployment)
+
+If you see a build or deployment running longer than expected, that may be one of the following cases:
+
+1. The build is blocked by a process in your build hook.
+2. The deployment is blocked by a long running process in your deploy hook.
+3. The deployment is blocked by a long running cron job in the environment.
+4. The deployment is blocked by a long running cron job in the parent environment.
+
+To determine if your environment is being stuck in the build or the deployment, you should look at the build log available on the UI. When you see a log line like below, your application is built and the deployment is stuck:
+
+```
+Re-deploying environment w6ikvtghgyuty-drupal8-b3dsina.
+```
+
+Here are the things you should do when you see your build or deployment is stuck:
+
+For a blocked _build_ (when you don't find the `Re-deployment environment ...` line), you have to create a [support ticket](https://platform.sh/support) to let us kill it.
+
+When a _deployment_ is blocked, you should try the following:
+
+1. Use [SSH](/development/access-site.md) to connect to your environment. Find any long-running cron jobs on the environment by running `ps afx`. Once you have identified the long running process on the environment, kill it.
+2. If you're performing "Sync", "Merge", or "Activate" on an environment and the process is stuck, use [SSH](/development/access-site.md) to connect to the parent environment and identify any long running cron jobs with `ps afx`. Kill the job(s) if you see any.
+
+Stuck builds should automaticly be killed after one hour. If they are stuck for a longer period please create [support ticket](https://platform.sh/support).
+
+
+## Slow or failing build (or deployment)
+
+Builds that take long time or fail is a common problem. Most of the time it's related to an application issue and they can be hard to troubleshoot without guidance.
+
+Here are a few tips that can help you solve the issues you are experiencing.
+
+### Check for errors in the logs
+
+Invisible errors during the build and deploy phase can cause increased wait times, failed builds and other problems. Investigating each log and fixing errors is essential.
+
+Related documentation: [Accessing logs](https://docs.platform.sh/development/logs.html#accessing-logs)
+
+### Check php.access.log
+
+The php.access.log contains the page executions by php-fpm. It also includes the execution time and peak memory usage of each request. This can be used to find pages and scripts that use too much memory or time to finish.
```

Comment:
This should add a caveat about "if you're using PHP".  Presumably that file won't exist otherwise.

Reply:
Also, PHP-FPM is capitalized.